workflow/controller/operator.go

```diff
@@ -1362,6 +1366,20 @@ type executeTemplateOpts struct {
 	onExitTemplate bool
 }
 
+func getNodeType(tmplType wfv1.TemplateType) wfv1.NodeType {
+	switch tmplType {
+	case wfv1.TemplateTypeContainer, wfv1.TemplateTypeScript, wfv1.TemplateTypeResource:
+		return wfv1.NodeTypePod
+	case wfv1.TemplateTypeDAG:
+		return wfv1.NodeTypeDAG
+	case wfv1.TemplateTypeSteps:
+		return wfv1.NodeTypeSteps
+	case wfv1.TemplateTypeSuspend:
+		return wfv1.NodeTypeSuspend
+	}
+	return ""
+}
```

Comment:
I do think this code does not handle a template which has retries, because it will incorrectly identify it as Pod and not Retry.